File: 461. Hamming Distance/solution.py
class Solution:
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        d = 0
        a = x ^ y
        while a:
            b = a >> 1
            b = b << 1
            
            if a != b:
                d += 1
                
            a = a >> 1
        
        return d

File: 461. Hamming Distance/solution.java
class Solution {
    public int hammingDistance(int x, int y) {
        x = x ^ y;
        y = x;
        int count = 0;
        while (x != 0) {
            y = x;
            x = x>>1;
            x = x<<1;
            if (y != x) {
                count++;
            }
            x = x>>1;
        }
        return count;
    }
}
